src/main/com/fulcrologic/rad/database_adapters/sql/utils.clj

```diff
@@ -0,0 +1,36 @@
+(ns com.fulcrologic.rad.database-adapters.sql.utils
+  (:require
+    [camel-snake-kebab.core         :as csk]
+    [clojure.string                 :as str]
+    [com.fulcrologic.rad.attributes :as attr]))
+
+
+(defmacro alias!
```

Comment:
Meh.  I know these are clj-only files, but I hate to introduce manual aliasing like this...I wish they would fix the aliasing issue in Clojure already.  Such a "util" (and I dislike "util" nses...too general) only works in clj leads to misunderstandings, IDEs commonly don't "get it" so it breaks tooling, etc.  Frankly, I'd rather type them out than use an alias trick like this.

Another possible workaround is to use a ns for the plugin's keywords that are used in open maps that has *no* code in it, but exists purely for reference/aliasing purposes.  After all, all we need is *a* namespace in which to put our "owned keywords"...they don't have to go with the individual nses.

Thoughts?

Reply:
This is a thorn in my eye as well, believe me. I indeed only found it remotely acceptable because sql is clj only, but I agree that's just incidental. Clojure really should give us reader aliases, because it's getting unnecessarily difficult in modern clojure apps (eg long kws all over the place) to organize namespaces. Any way we could strengthen the interest in https://clojure.atlassian.net/browse/CLJ-2123 ?

Anyway I had already thought about your suggestions. Writing them out is the most practical solution, but also verbose and... I'm quite a neat freak in case you haven't noticed 😉.

The empty namespace one is my favourite, good place to put specs. I realise that the only reason I wanted to do it this way is because I wanted the namespace that holds the keys for the public API of this plugin also be the namespace that holds the functions that users will call. I think for now I'm gonna go with having the user require the different implementation namespaces for using the functions they want. It would make the require lists more verbose but maybe for now that's the best way.

Reply:
Oh an you're right about the utils ns. Taking a step back I realise this ns should be about the sql schema and questions about it.